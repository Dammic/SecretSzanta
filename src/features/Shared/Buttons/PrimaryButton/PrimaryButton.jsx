import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { Button } from '..'

import styles from './PrimaryButton.css'

const PrimaryButton = (props) => {
    const { className, children } = props
    return (
        <Button
            {...props}
            className={classNames(styles.primary, className)}
        >
            {children}
        </Button>
    )
}

PrimaryButton.propTypes = {
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    className: PropTypes.string,
}

export default PrimaryButton
