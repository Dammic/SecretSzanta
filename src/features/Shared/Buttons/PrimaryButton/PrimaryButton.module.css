@import 'variables/variables.css';

.primary {
    background-color: var(--button-primary-bg);
    color: var(--button-primary-font-color);

    &:hover {
        background-color: var(--button-primary-bg-hover);
    }
}
