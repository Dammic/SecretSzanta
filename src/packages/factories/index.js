export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
