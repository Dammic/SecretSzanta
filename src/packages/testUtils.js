import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
