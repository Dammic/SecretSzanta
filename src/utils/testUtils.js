import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
