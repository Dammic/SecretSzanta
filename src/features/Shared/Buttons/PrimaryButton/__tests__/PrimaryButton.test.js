import React from 'react'
import { shallow } from 'enzyme'
import { expectMatchingSnapshot } from 'packages/testUtils'
import PrimaryButton from '../PrimaryButton'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        children: <span>mockLabel</span>,
        className: 'mockClassName',
        ...propsOverrides,
    }
    const component = renderMethod(<PrimaryButton {...props} />)
    return { props, component }
}

describe('<PrimaryButton />', () => {
    it('matches snapshot', () => {
        const { props } = setupProps()
        expectMatchingSnapshot(<PrimaryButton {...props} />)
    })
})
