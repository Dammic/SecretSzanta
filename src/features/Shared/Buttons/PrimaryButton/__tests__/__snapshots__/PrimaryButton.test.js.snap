// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> matches snapshot 1`] = `
<div
  className="btn primary mockClassName"
  role="button"
  tabIndex="0"
>
  <span>
    mockLabel
  </span>
</div>
`;
